dotnet/Cucumber.HtmlFormatter/MessagesToHtmlWriter.cs

```diff
@@ -6,23 +6,46 @@ namespace Cucumber.HtmlFormatter;
 public class MessagesToHtmlWriter : IDisposable
 {
     private StreamWriter writer;
+    private Func<StreamWriter, Envelope, Task> asyncStreamSerializer;
     private Action<StreamWriter, Envelope> streamSerializer;
     private string template;
     private JsonInHtmlWriter JsonInHtmlWriter;
     private bool streamClosed = false;
     private bool preMessageWritten = false;
     private bool firstMessageWritten = false;
     private bool postMessageWritten = false;
+    private bool isAsyncInitialized = false;
 
+    [Obsolete("Cucumber.HtmlFormatter moving to async only operations. Please use the MessagesToHtmlWriter(Stream, Func<StreamWriter, Envelope, Task>) constructor", false)]
     public MessagesToHtmlWriter(Stream stream, Action<StreamWriter, Envelope> streamSerializer) : this(new StreamWriter(stream), streamSerializer)
     {
     }
+    public MessagesToHtmlWriter(Stream stream, Func<StreamWriter, Envelope, Task> asyncStreamSerializer) : this(new StreamWriter(stream), asyncStreamSerializer) { }
+
+    [Obsolete("Cucumber.HtmlFormatter moving to async only operations. Please use the MessagesToHtmlWriter(StreamWriter, Func<StreamWriter, Envelope, Task>) constructor", false)]
```

Comment:
This looks like a deprecation. So this needs its own `deprecated` entry in the changelog. This will inform the decision for the next version number.